### PR TITLE
[Develop] chore: [sc-32325] Migrate pd attachments page to unicef_attachments

### PR DESCRIPTION
**Pull request by roman-karpovich**
Story details: https://app.shortcut.com/unicefetools/story/32325

### CODE REVIEW

**Comment by roman-karpovich** [comment]
@robertavram @domdinicola could we please release updated version of unicef_attachments? PR depends on new migration from there.
https://github.com/unicef/unicef-attachments/pull/37

**Comment by robertavram** [comment]
https://pypi.org/project/unicef-attachments/0.12/

File: src/etools/applications/partners/serializers/interventions_v2.py [hunk 1]
```diff
@@ -397,21 +396,46 @@ class Meta:
...
+        return url
+
+    def to_internal_value(self, data):
+        return data
+
+
+class InterventionAttachmentSerializer(serializers.ModelSerializer):
+    id = serializers.IntegerField(read_only=True)
+    intervention = serializers.IntegerField(read_only=True, source='object_id')
+    attachment = serializers.FileField(read_only=True, source='file')
```
Comment (emaciupe) [inline src/etools/applications/partners/serializers/interventions_v2.py:421]: @roman-karpovich attachment and attachment_file are identical here - was attachment meant to be a pk? like in PMPInterventionAttachmentSerializer?

File: src/etools/applications/partners/views/interventions_v3.py [hunk 2]
```diff
@@ -572,64 +573,63 @@ def get_queryset(self):
...
         super().perform_update(serializer)
         self.perform_auto_transitions(self.get_root_object(), self.request.user)
 
+    def delete(self, request, *args, **kwargs):
+        intervention = self.get_root_object()
+
+        if intervention.status in [Intervention.DRAFT]:
+            return super().delete(request, *args, **kwargs)
+        else:
+            raise ValidationError("Deleting an attachment can only be done in Draft status")
```
Comment (emaciupe) [inline src/etools/applications/partners/views/interventions_v3.py:631]: @roman-karpovich please mark the validation message as translatable _("")

File: src/etools/applications/partners/views/interventions_v2.py [hunk 3]
```diff
@@ -345,32 +345,45 @@ def update(self, request, *args, **kwargs):
...
+        return queryset
+
     def delete(self, request, *args, **kwargs):
         obj = self.get_object()
 
-        if obj.intervention.status in [Intervention.DRAFT]:
+        if obj.content_object.status in [Intervention.DRAFT]:
             return super().delete(request, *args, **kwargs)
         else:
             raise ValidationError("Deleting an attachment can only be done in Draft status")
```
Comment (emaciupe) [inline src/etools/applications/partners/views/interventions_v2.py:389]: @roman-karpovich can you also prefix with translatable? seems i missed it. (for views use `from django.utils.translation import gettext as _`)

**Review by emaciupe — CHANGES_REQUESTED** [review]

File: src/etools/applications/partners/serializers/interventions_v2.py (hunk 1 above)
Comment (roman-karpovich) [reply src/etools/applications/partners/serializers/interventions_v2.py:421]: updated

File: src/etools/applications/partners/views/interventions_v3.py (hunk 2 above)
Comment (roman-karpovich) [reply src/etools/applications/partners/views/interventions_v3.py]: updated

File: src/etools/applications/partners/views/interventions_v2.py (hunk 3 above)
Comment (roman-karpovich) [reply src/etools/applications/partners/views/interventions_v2.py]: updated

**Comment by emaciupe** [comment]
@roman-karpovich I will remove the .po files line numbering from the translations so there will be a lot less failing on test checks

**Comment by roman-karpovich** [comment]
> @roman-karpovich I will remove the .po files line numbering from the translations so there will be a lot less failing on test checks

agreed. now it's better to revert those changes. localisations were broken at some moment, so i regenerated them. now they are fine in develop

**Review by emaciupe — APPROVED** [review]
Looks good!

**Pull request closed without merging**